examples/dbscan/ArborX_DBSCAN.hpp

```diff
@@ -236,60 +236,80 @@ void dbscan(ExecutionSpace exec_space, Primitives const &primitives,
   start = clock::now();
   Kokkos::Profiling::pushRegion("ArborX::DBSCAN::sort_and_filter_clusters");
 
+  Kokkos::View<int *, MemorySpace> cluster_sizes(
+      "ArborX::DBSCAN::cluster_sizes", n);
+  Kokkos::parallel_for("ArborX::DBSCAN::compute_cluster_sizes",
+                       Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
+                       KOKKOS_LAMBDA(int const i) {
+                         Kokkos::atomic_fetch_add(&cluster_sizes(clusters(i)),
+                                                  1);
+                       });
+
+  // The idea here is to replace cluster indices for small clusters (containing
+  // less than cluster_min_size points) to INT_MAX. This way, during the sort
+  // routine afterwards, they will be at the end of the permutation array,
+  // allowing us to simply truncate it to get the result.
+  int num_skipped = 0;
+  Kokkos::parallel_reduce("ArborX::DBSCAN::replace_skipped_cluster_indices",
+                          Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
+                          KOKKOS_LAMBDA(int const i, int &update) {
+                            if (cluster_sizes(clusters(i)) < cluster_min_size)
+                            {
+                              clusters(i) = INT_MAX;
+                              update++;
+                            }
+                          },
+                          num_skipped);
+  auto num_cluster_indices = n - num_skipped;
+
   // sort clusters and compute permutation
   auto permute = Details::sortObjects(exec_space, clusters);
 
-  reallocWithoutInitializing(cluster_offset, n + 1);
+  // truncate the permutation array, see comment above
+  reallocWithoutInitializing(cluster_indices, num_cluster_indices);
+  Kokkos::deep_copy(
+      exec_space, cluster_indices,
+      Kokkos::subview(permute, std::make_pair(0, num_cluster_indices)));
```

Comment:
This is a `Kokkos::resize()`isn't it?  Did you chose do it this way because Kokkos initializes the array before copying entries?  If so please comment.

If I understand it right, this is something that could be reported to Kokkos.

Reply:
Thanks for bringing this to my attention. Now that I look at the complete code, I realize that we no longer use `permute` in any of the following kernels, and thus don't have to be careful with it anymore. Thus, you are right, it could be done as
```c++
Kokkos::resize(permute, num_cluster_indices));
cluster_indices = permute;
```
At this moment, I'm not sure about the complexity of the first line. Is it 1 allocation, 1 initialization of size `num_cluster_indices` and 1 deep_copy? If yes, that it indeed has an extra initialization cost that we don't need. I think I tried a solution proposed in [this](https://github.com/kokkos/kokkos/issues/2048#issuecomment-528034887) comment (`Kokkos::resize( Kokkos::view_resize (WithoutInitializing), A, 2*N );`) but it did not compile, because there is no `view_resize`.

Reply:
Actually, it seems to have been fixed in kokkos/kokkos#2285. So it almost works, except they have different value types:
```
error: no operator "=" matches these operands
            operand types are:
  Kokkos::View<int *, Kokkos::Cuda::memory_space> =
  Kokkos::View<unsigned int *, Kokkos::Device<Kokkos::CudaSpace::execution_space, Kokkos::CudaSpace::memory_space>>
```

Reply:
Try `Kokkos::resize(Kokkos::WithoutInitializing, A, 2*N);

Reply:
Anyway I missed that you were copying into another view.